### PR TITLE
Fix ability to refresh token on API calls

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

This PR fixes the issue when API calls use an expired token. It improves the application pre-start set up so the `che-workspace-client` gets initialized in the correct order, after the keycloak adapter setup. That allows `che-workspace-client` to correctly register a request interceptor that does token refreshing.

### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/19058

### CODE REVIEW

File: src/services/cheWorkspaceClient/index.ts [hunk 1]
```diff
@@ -53,39 +56,30 @@ export class CheWorkspaceClient {
...
 
-    window.addEventListener('message', (event: MessageEvent) => {
-      if (event.data.startsWith('update-token:')) {
-        const receivedValue = parseInt(event.data.split(':')[1], 10);
-        const validityTime = Number.isNaN(receivedValue) ? VALIDITY_TIME : Math.ceil(receivedValue / 1000);
-        this.handleRefreshToken(validityTime);
-      }
-    }, false);
+      this.axios.interceptors.request.use(async config => {
+        console.log('>>> interceptor');
```
Comment (sleshchenko) [inline src/services/cheWorkspaceClient/index.ts:65]:
```suggestion
```

**Review by sleshchenko — COMMENTED** [review]
I even did not try to review the code, but let me just left <3 here for solving issue with using expired token.

According to discussion we had, it's going to refresh token before each request, which is not the most optimal way, but I'm OK with that since it's really stable and you told old dashboard did in the same way.
We can optimize it later if needed.

I'll test it a bit later today

**Comment by akurinnoy** [comment]
@sleshchenko I've looked into token refreshing and found out that requests for updating token are sent only when it's necessary and the token is nearly to expire, not each time when an update method is called. So we don't have to optimize this part.

**Review by olexii4 — APPROVED** [review]
LGTM

**Review by JPinkney — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
[dashboard-next-ci-test]

**Comment by dmytro-ndp** [comment]
[dashboard-next-ci-test]

**Comment by sleshchenko** [comment]
> @sleshchenko I've looked into token refreshing and found out that requests for updating token are sent only when it's necessary and the token is nearly to expire, not each time when an update method is called. So we don't have to optimize this part.

That's great news!!!
Can we update token forcibly before Start A Workpace request?
It could help Che Server to minimize a chance to use expired token during workspace start process. If it's not done yet, I did not tested or checked the code.

**Comment by dmytro-ndp** [comment]
[dashboard-next-ci-test]

**Pull request merged**